src/snowflake/snowpark/_internal/analyzer/analyzer.py

```diff
@@ -166,18 +167,33 @@ def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
         self.generated_alias_maps = {}
         self.subquery_plans = []
         self.alias_maps_to_use: Optional[Dict[uuid.UUID, str]] = None
+        self.config_context: ConfigContext = ConfigContext(session)
+        # Point this plan builder to the same config context as the analyzer
+        self.plan_builder.config_context = self.config_context
 
     def analyze(
         self,
         expr: Union[Expression, NamedExpression],
         df_aliased_col_name_to_real_col_name: DefaultDict[str, Dict[str, str]],
         parse_local_name=False,
+    ) -> str:
+        # Set the config context for analysis step
+        with self.config_context:
+            return self.do_analyze(
+                expr, df_aliased_col_name_to_real_col_name, parse_local_name
+            )
+
+    def do_analyze(
```

Comment:
another thing here is that caller can still call do_analyze independently if they want, and this may not have a snapshot

Reply:
`do_analyze` is only called from within Analyzer similar to `do_resolve`. I can convert them into `_do_analyze` to make this a private method to ensure a snapshot is created by calling `analyze()`